tests/app/v2/broadcast/test_post_broadcast.py

```diff
@@ -119,6 +119,126 @@ def test_valid_post_cap_xml_broadcast_returns_201(
+):
+    auth_header = create_service_authorization_header(service_id=sample_broadcast_service.id)
+
+    # create a broadcast
+    response_for_create = client.post(
+        path='/v2/broadcast',
+        data=sample_cap_xml_documents.WAINFLEET,
+        headers=[('Content-Type', 'application/cap+xml'), auth_header],
+    )
+    assert response_for_create.status_code == 201
+
+    response_json_for_create = json.loads(response_for_create.get_data(as_text=True))
+
+    assert response_json_for_create['cancelled_at'] is None
+    assert response_json_for_create['cancelled_by_id'] is None
+    assert response_json_for_create['reference'] == '50385fcb0ab7aa447bbd46d848ce8466E'
+    assert response_json_for_create['status'] == 'pending-approval'
+
+    # cancel broadcast
+    response_for_cancel = client.post(
+        path='/v2/broadcast',
+        data=sample_cap_xml_documents.WAINFLEET_CANCEL,
+        headers=[('Content-Type', 'application/cap+xml'), auth_header],
+    )
+    assert response_for_cancel.status_code == 201
+
+    response_json_for_reject = json.loads(response_for_cancel.get_data(as_text=True))
+
+    assert response_json_for_reject['reference'] == response_json_for_create['reference']
+    assert response_json_for_reject['status'] == 'rejected'
+    assert response_json_for_reject['updated_at'] is not None
+
+
+def test_valid_cancel_broadcast_request_cancels_active_alert_and_returns_201(
+    client,
+    sample_broadcast_service,
+    notify_db,
+    notify_db_session,
+    mocker
+):
+    auth_header = create_service_authorization_header(service_id=sample_broadcast_service.id)
+
+    # create a broadcast
+    response_for_create = client.post(
+        path='/v2/broadcast',
+        data=sample_cap_xml_documents.WAINFLEET,
+        headers=[('Content-Type', 'application/cap+xml'), auth_header],
+    )
+    assert response_for_create.status_code == 201
+
+    response_json_for_create = json.loads(response_for_create.get_data(as_text=True))
+
+    assert response_json_for_create['cancelled_at'] is None
+    assert response_json_for_create['cancelled_by_id'] is None
+    assert response_json_for_create['reference'] == '50385fcb0ab7aa447bbd46d848ce8466E'
+    assert response_json_for_create['status'] == 'pending-approval'
+
+    # approve broadcast
+    broadcast_message = dao_get_broadcast_message_by_id_and_service_id(
+        response_json_for_create["id"], response_json_for_create["service_id"]
+    )
+    broadcast_message.status = 'broadcasting'
+
+    # cancel broadcast
+    mock_task = mocker.patch('app.celery.broadcast_message_tasks.send_broadcast_event.apply_async')
+
+    response_for_cancel = client.post(
+        path='/v2/broadcast',
+        data=sample_cap_xml_documents.WAINFLEET_CANCEL,
+        headers=[('Content-Type', 'application/cap+xml'), auth_header],
+    )
+    assert response_for_cancel.status_code == 201
+
+    assert len(broadcast_message.events) == 1
+    alert_event = broadcast_message.events[0]
```

Comment:
I think since I plan to move this part of the code to a utils file and test it separately, we will not need extra test for this in the API flow (since it will be the common util part that deals with this, and it will be tested there, at source)